pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -1075,14 +1093,19 @@ def generateDrawSpecs(self, p):
             ## Get the list of strings to display for this level
             if tickStrings is None:
                 spacing, values = tickLevels[i]
-                strings = self.tickStrings(values, self.autoSIPrefixScale * self.scale, spacing)
+
+                strings = list(self.tickStrings(
```

Comment:
Yeah so as mentioned in comments, ideally I'd like to strip out the `list` casting since I don't really think we benefit much from the *initial pass* style of *avoiding generating ticks for graphics not in view* that's going on here. 

Further even if we do want to keep that, after the initial generation avoiding re-generating is mostly fruitless (especially with caching) afaict unless someone is expecting a dynamic domain 😂

I dunno though, maybe that's a use case?